assets/src/components/stacks/state/StackStateGraph.tsx

```diff
@@ -9,55 +9,96 @@ import { ReactFlowGraph } from '../../utils/reactflow/ReactFlowGraph'
 
 import { LayoutOptions } from 'elkjs'
 import { StackStateGraphNode } from './StackStateGraphNode'
+import { Flex, Input, SearchIcon } from '@pluralsh/design-system'
+import Fuse from 'fuse.js'
+import { isEmpty } from 'lodash'
 
 const nodeTypes = {
   [NodeType.Stage]: StackStateGraphNode,
 }
 
-function getNodesAndEdges(state: StackState) {
+const searchOptions = {
+  keys: [
+    'name',
+    {
+      name: 'configuration',
+      getFn: (r) => JSON.stringify(r?.configuration ?? {}),
```

Comment:
It looks like fuse.js does not offer search on all object fields so this has to be transformed.